plugins/helpers.nix

```diff
@@ -1,8 +1,49 @@
-{ lib, ... }:
+{ lib, config, ... }:
 with lib;
 rec {
+
+  boolOption = default: description: mkOption {
+    type = types.bool;
+    description = description;
+    default = default;
+  };
+
+  intOption = default: description: mkOption {
+    type = types.int;
+    description = description;
+    default = default;
+  };
+
+  strOption = default: description: mkOption {
+    type = types.str;
+    description = description;
+    default = default;
+  };
+
+  boolNullOption = description: mkOption {
+    type = types.nullOr types.bool;
+    description = description;
+    default = null;
+  };
+
+  intNullOption = description: mkOption {
+    type = types.nullOr types.int;
+    description = description;
+    default = null;
+  };
+
+  strNullOption = description: mkOption {
+    type = types.nullOr types.str;
+    description = description;
+    default = null;
+  };
+
+  typeOption = type: default: description: mkOption {
+    inherit type default description;
+  };
+
   # vim dictionaries are, in theory, compatible with JSON
-  toVimDict = args: toJSON 
+  toVimDict = args: toJSON
     (lib.filterAttrs (n: v: !isNull v) args);
 
   # Black functional magic that converts a bunch of different Nix types to their
```

Comment:
just a note for me, but forgot to complete this sentence lol, "to their vimscript equivalents"